### PR TITLE
Miscellaneous fixes for MobileNet

### LINKED ISSUE #178: MobileNetv2/3 need padding in first layer
I think to comply with the papers [v2](https://arxiv.org/pdf/1801.04381.pdf) Table 2 / [v3](https://arxiv.org/pdf/1905.02244.pdf) Table 1, the first conv layer in the respective models needs pad=1. Easy fix, but dont have time to make a PR right now. Also this allows better composability as UNet backbone in downstream packages.

### PR DESCRIPTION
**Pull request by theabhirath**
Closes #178. Also removes an extra fully connected layer in MobileNetv1, making it now conform to the configuration described in the paper. And some minor docs and formatting changes I caught have also made it through

### CODE REVIEW

**Comment by theabhirath** [comment]
This is also another step towards #176, as all the MobileNet models now expose `inchannels` and `classes`

**Review by darsnack — CHANGES_REQUESTED** [review]

File: src/convnets/mobilenet.jl [hunk 1]
```diff
@@ -46,12 +44,11 @@ function mobilenetv1(width_mult, config;
     return Chain(Chain(layers),
                  Chain(GlobalMeanPool(),
                        MLUtils.flatten,
-                       Dense(inchannels, fcsize, activation),
-                       Dense(fcsize, nclasses)))
+                       Dense(inchannels, nclasses)))
```
[the remark this answers is not in the document]
Comment (theabhirath) [reply src/convnets/mobilenet.jl:47]: I feel liked we are going to be making a _huge_ breaking release anyways sometime after my GSoC work lands, so this just gets added to the list

File: src/convnets/convmixer.jl [hunk 2]
```diff
@@ -9,7 +9,7 @@ Creates a ConvMixer model.
 
   - `planes`: number of planes in the output of each block
   - `depth`: number of layers
-  - `inchannels`: number of channels in the input
+  - `inchannels`: The number of channels in the input. The default value is 3.
```
[the remark this answers is not in the document]
Comment (theabhirath) [reply src/convnets/convmixer.jl:12]: I'm working on making this happen for all the docstrings in the new PRs (that is, having the default values in there). I need to push a larger documentation change sometime in the next few weeks and that should probably cover everything

**Comment by darsnack** [comment]
Yes, this is the main reason I brought it up. I want to make a patch release with the recent changes before merging the breaking overhaul.

**Comment by theabhirath** [comment]
The breaking change has been undone for now. The rest I think should fit into the patch-release without a problem

File: src/convnets/mobilenet.jl [hunk 3]
```diff
@@ -4,8 +4,7 @@
     mobilenetv1(width_mult, config;
                 activation = relu,
                 inchannels = 3,
-                nclasses = 1000,
-                fcsize = 1024)
+                nclasses = 1000)
```
Comment (darsnack) [inline src/convnets/mobilenet.jl:7]:
```suggestion
                fcsize = 1024,
                nclasses = 1000)
```

File: src/convnets/mobilenet.jl [hunk 4]
```diff
@@ -21,23 +20,24 @@ Create a MobileNetv1 model ([reference](https://arxiv.org/abs/1704.04861v1)).
...
+  - `inchannels`: The number of input channels. The default value is 3.
   - `fcsize`: The intermediate fully-connected size between the convolution and final layers
   - `nclasses`: The number of output classes
 """
 function mobilenetv1(width_mult, config;
                      activation = relu,
                      inchannels = 3,
-                     nclasses = 1000,
-                     fcsize = 1024)
+					 fcsize = 1024,
```
Comment (darsnack) [inline src/convnets/mobilenet.jl:30]:
```suggestion
					     fcsize = 1024,
```

**Review by darsnack — CHANGES_REQUESTED** [review]

**Review by darsnack — APPROVED** [review]

File: src/convnets/mobilenet.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (darsnack) [reply src/convnets/mobilenet.jl]: Or (user) code that relies on the structure but doesn't use Functors.jl. I think previously we decided to consider structural changes as breaking. But the more nuanced question here is whether a structural change that fixes an error is considered a bug fix.

I think probably here no one would complain given how recent the model definitions are.

**Comment by darsnack** [comment]
Can you rebase with the test changes from #171?

**Comment by theabhirath** [comment]
....wait. How did the CI go from one check passing to _all_ checks passing

**Comment by ToucheSir** [comment]
Something to do with #171 disabling tests for some of the largest models?

**Comment by theabhirath** [comment]
Oh, possibly 🤦🏽 I feel like this is just 50-50 at this point as to whether the tests pass or not

**Comment by ToucheSir** [comment]
Looks like the failure was a timeout instead of an OOM, so hopefully it was an anomaly!

**Comment by theabhirath** [comment]
Should be ready for merge and release now 😄

**Comment by maxfreu** [comment]
Thanks!

**Pull request merged**